src/stats/dendrogram.jl

```diff
@@ -0,0 +1,178 @@
+# Node struct
+struct DNode{N}
+    idx::Int
+    position::Point{N, Float32}
+    children::Union{Tuple{Int,Int}, Nothing}
+end
+
+function DNode(idx::Int, point::Point{N}, children::Union{Tuple{Int,Int}, Nothing}) where N
+    return DNode{N}(idx, point, children)
+end
+
+"""
+    dendrogram(x, y; kwargs...)
+
+Draw a [dendrogram](https://en.wikipedia.org/wiki/Dendrogram),
+with leaf nodes specified by `x` and `y` coordinates,
+and parent nodes identified by `merges`.
+# Arguments
+- `x`: x positions of leaf nodes
+- `y`: y positions of leaf nodes (default = 0)
+# Keywords
+- `merges`: specifies connections between nodes (see below)
+- `treestyle`: one of `:tree`, `:box`.  Overload `dendrogram_connectors(::Val{:mystyle}, parent, child1, child2)` to define a new style.
```

Comment:
These need to be updated, and I don't see an equivalent to `merges` anywhere here?

Reply:
`merges` should come from the Hclust object itself I think - it's a vector of tuples specifying the order of leaves

https://juliastats.org/Clustering.jl/stable/hclust.html#Clustering.Hclust